esp/esdllib/esdl_xpath_extensions_libxml.cpp

```diff
@@ -176,7 +189,175 @@ static void getFeatureSecAccessFlagsFunction (xmlXPathParserContextPtr ctxt, int
     xmlXPathReturnNumber(ctxt, access);
 }
 
-void registerEsdlXPathExtensions(IXpathContext *xpathContext, IEspContext *context, const StringArray &prefixes)
+/**
+ * getStoredStringValueFunction
+ * @ctxt:  an XPath parser context
+ * @nargs:  the number of arguments
+ *
+ */
+static void getStoredStringValueFunction (xmlXPathParserContextPtr ctxt, int nargs)
+{
+    IEsdlScriptContext *scriptContext = getEsdlScriptContext(ctxt);
+    if (!scriptContext)
+    {
+        xmlXPathSetError((ctxt), XPATH_INVALID_CTXT);
+        return;
+    }
+
+    if (nargs != 1)
+    {
+        xmlXPathSetArityError(ctxt);
+        return;
+    }
+
+    xmlChar *namestring = xmlXPathPopString(ctxt);
+    if (xmlXPathCheckError(ctxt)) //includes null check
+        return;
+
+    const char *value = scriptContext->queryAttribute("store", (const char *)namestring);
+    xmlFree(namestring);
+    if (!value)
+        xmlXPathReturnEmptyString(ctxt);
+    else
+        xmlXPathReturnString(ctxt, xmlStrdup((const xmlChar *)value));
+}
+
+/**
+ * getLogOptionFunction
+ * @ctxt:  an XPath parser context
+ * @nargs:  the number of arguments
+ *
+ */
+static void getLogOptionFunction (xmlXPathParserContextPtr ctxt, int nargs)
+{
+    IEsdlScriptContext *scriptContext = getEsdlScriptContext(ctxt);
+    if (!scriptContext)
+    {
+        xmlXPathSetError((ctxt), XPATH_INVALID_CTXT);
+        return;
+    }
+
+    if (nargs != 1)
+    {
+        xmlXPathSetArityError(ctxt);
+        return;
+    }
+
+    xmlChar *namestring = xmlXPathPopString(ctxt);
+    if (xmlXPathCheckError(ctxt)) //includes null check
+        return;
+
+    const char *value = scriptContext->queryAttribute("logging", (const char *)namestring);
```

Comment:
This comment applies to multiple functions. The literal, or literals elsewhere, should be using the defined names.